action.yml

```diff
@@ -29,27 +29,60 @@ outputs:
 runs:
   using: composite
   steps:
+    - id: registryconfig
+      if: ${{ cloudbees.registries != '' }}
+      name: Resolve source image
+      uses: docker://020229604682.dkr.ecr.us-east-1.amazonaws.com/services/registry-config:0.0.22
+      with:
+        entrypoint: /registry-config
+        args: --config="${{ cloudbees.registries }}" resolve "${{ inputs.src }}" resolved-image.file
+
     - id: cranecopy
       name: Copy a remote image from a given source to a destination
       uses: docker://gcr.io/go-containerregistry/crane:debug
       run: |
-        set -euo pipefail
+        set -uo pipefail
 
-        if [ "$SKIP_IMAGE_VALIDATION" == false ]; then
-          echo "Validating source image ${{ inputs.src }}"
-          crane validate --remote ${{ inputs.src }} --platform ${{ inputs.platform }}
+        if [ ! -e resolved-image.file ]; then
+          echo '${{ inputs.src }}' > resolved-image.file
         fi
 
-        echo "Copying source image ${{ inputs.src }} to destination ${{ inputs.destination }}"
-        crane copy ${{ inputs.src }} ${{ inputs.destination }} --platform ${{ inputs.platform }}
+        set +e
+
+        while IFS= read -r line; do
+          if [ "$SKIP_IMAGE_VALIDATION" == false ]; then
+            echo "Validating source image ${line}"
+            crane validate --remote ${line} --platform ${{ inputs.platform }}
+
+            if [ $? -ne 0 ]; then
+              echo "Failed to validate image: $line"
+              continue
+            fi
+          fi
+
+          echo "Copying source image ${line} to destination ${{ inputs.destination }}"
+          crane copy ${line} ${{ inputs.destination }} --platform ${{ inputs.platform }}
+
+          if [ $? -ne 0 ]; then
+            echo "Failed to copy image: $line"
+            continue
+          fi
+
+          source_image=$line
+          break
+        done < "resolved-image.file"
+        
+        rm resolved-image.file
 
+        set -e
+        
         if [ "$SKIP_IMAGE_VALIDATION" == false ]; then
           echo "Validating destination image ${{ inputs.destination }}"
           crane validate --remote ${{ inputs.destination }} --platform ${{ inputs.platform }}
         fi
 
-        echo "Getting image digest of source image ${{ inputs.src }}"
-        crane digest ${{ inputs.src }} --full-ref --platform ${{ inputs.platform }}
+        echo "Getting image digest of source image ${source_image}"
+        crane digest ${source_image} --full-ref --platform ${{ inputs.platform }}
```

Comment:
While your PR didn't cause the problem I realize the Action has a race condition here because it obtains the digest after the image was copied at which point the image could have been changed again by another concurrently running process/workflow/job.
While solving that exceeds the scope of your ticket, it would be great if you could fix that while you're on it or capture that within a new Jira ticket: The source image digest should be resolved first and then the image copied using the digest.

Reply:
Not sure why we even need this line. My understanding is that it just for logging purposes, No? Also moving it before not solving problem as it still can be update between this line and copy operation.

Reply:
Updated according to our conversation.